src/main/groovy/jaci/gradle/deploy/cache/NeedsUpdateFunction.groovy

```diff
@@ -0,0 +1,10 @@
+package jaci.gradle.deploy.cache
+
+import groovy.transform.CompileStatic
+import jaci.gradle.deploy.context.DeployContext
+
+@CompileStatic
+@FunctionalInterface
+interface NeedsUpdateFunction {

```

Comment:
Just as a nitpick, I'd probably call this `CacheChecker` or something to that degree

Reply:
Agreed. Fixed